pkg/datastore/datastore.go

```diff
@@ -61,6 +61,25 @@ var (
 	ErrUnimplemented   = errors.New("unimplemented")
 )
 
+type Writer string
+
+const (
+	// UnknownWriter indicates that the writer is unknown.
+	// This writer is unable to perform any Update operation.
+	UnknownWriter Writer = "unknown"
+	// TestWriter indicates that the writer is test.
+	// This writer is dedicated for testing.
+	TestWriter Writer = "test"
+	// WebWriter indicates that the writer is web.
+	WebWriter Writer = "web"
+	// PipectlWriter indicates that the writer is pipectl.
+	PipectlWriter Writer = "pipectl"
```

Comment:
I wonder how we will merge fields of the model if these writers are different ones. 🤔

Reply:
Not really get your point, but do you wonder about just these two writers web and pipectl or between any pair from those writers? For the first case, I think basically we can merge fields which be writen by these two writers to one, I just want to separte the source here to enable us change the logic of merging field by writer or something like that later, the writer and where it can write to should be define separately. For the latter, I think we're going to narrow the scope of each field's writer so that we will only have one writer for each, just like we did with the application's description field, were we? 🤔

Reply:
> we can merge fields which be writen by these two writers to store in a single file, I just want to separte the source here to enable us change the logic of merging field by writer or something like that later, the writer and where it can write to should be define separately

That is a good idea. I was thinking that the writer's name will be the prefix of the object file.
Now I got your idea and totally agree with it.

Reply:
Btw, I would like to find better names for them since `UnknownWritter` and `PipectlWriter` are a little bit hard to understand.

I propose to change from `Writer` to `Actor` or `Owner`. `Actor` means who is responsible for store actions.

So we can have

`UnknownWriter` -> `ReadOnlyActor`
`PipectlWriter` -> `APIActor` or `ExternalAPIActor` (API is not just used by pipectl but any external service`

What do you think?

Reply:
`Commander` looks good to me. Let me use it in my next discussions.

From my point of view, I see that just 4 commanders are enough:`PipedCommander`, `APICommander`, `OpsCommander`, `TestCommander`.

I still don't understand the reason for having the `Unknown` one since HTTP API can use `APICommander` as `web` and `pipectl`.
🤔

Reply:
I get your point 👍  Maybe, because I just want to specify who is the commander, that's why I create `WebCommander` and `PipectlCommander` separately, and with that kind of commanders created, I can't really tell what is the commander at the time I use that store object in HTTP API handler.

On my second thought, I think I may use the `WebCommander` at the place I use the `UnknownCommander` currently, and we will have 5 commanders: Piped, Ops, Web, Pipectl, Test. I recommend keeping `Web` and `Pipectl` separately and making things like "web and pipectl writeable fields are stored in API_XXX.json" because I guess that name like `APICommander` is a bit odd and common (hard to separate it with piped things - we have piped_api as well), and of course, the idea of separating who is the commander with what it can do has remained. wdyt?

Reply:
>  we will have 5 commanders: Piped, Ops, Web, Pipectl, Test

Ok. Let's go with this.

Reply:
Thanks for your comments 🙏  Addressed by https://github.com/pipe-cd/pipecd/pull/3243/commits/2194160e566bd175399323f3312e0e8512714801